src/olympia/search/filters.py

```diff
@@ -124,6 +125,29 @@ class AddonStatusFilterParam(AddonFilterParam):
     es_field = 'status'
 
 
+class AddonCategoryFilterParam(AddonFilterParam):
+    query_param = 'category'
+    es_field = 'category'
+    valid_values = CATEGORIES_BY_ID.keys()
+
+    def __init__(self, request):
+        super(AddonCategoryFilterParam, self).__init__(request)
+        # Category slugs are only unique for a given type+app combination.
+        # Once we have that, it's just a matter of selecting the corresponding
+        # dict in the categories constants and use that as the reverse dict,
+        # and make sure to use get_value_from_object_from_reverse_dict().
+        try:
+            app = AddonAppFilterParam(self.request).get_value()
+            type_ = AddonTypeFilterParam(self.request).get_value()
```

Comment:
`type` is a builtin ; overriding it works fine, but then it messes up syntax highlighting (including in github diffs) so I usually append a `_` to avoid it.